apps/overige-objecten-api/src/controllers/objects/create.ts

```diff
@@ -0,0 +1,85 @@
+import type { RequestHandler } from 'express';
+import snakeCase from 'lodash.snakecase';
+import { v4 } from 'uuid';
+import { CREATE_VAC } from '../../queries';
+import type { DataVacItem } from '../../strapi-product-type';
+import type { components } from '../../types/openapi';
+import { fetchData, getCurrentTypeParam, getTheServerURL } from '../../utils';
+
+type VACData = {
+  data: {
+    createVac: {
+      data: DataVacItem;
+    };
+  };
+};
+
+export const createVacController: RequestHandler = async (req, res, next) => {
+  const isAuthHasToken = req.headers?.authorization?.startsWith('Token');
+  const tokenAuth = isAuthHasToken ? req.headers?.authorization?.split(' ')[1] : req.headers?.authorization;
+  const locale = req.query?.locale || 'nl';
+  const graphqlURL = new URL('/graphql', process.env.STRAPI_PRIVATE_URL);
+  const serverURL = getTheServerURL(req);
+
+  try {
+    const body = req.body as components['schemas']['ObjectData'];
+    const vac = body?.record?.data as components['schemas']['vac'];
+    const vacSchemaURL = new URL('api/v2/objecttypes/vac', serverURL).href;
+    const { isVac } = getCurrentTypeParam(body.type);
+
+    if (!isVac) {
+      return res.status(400).json({ message: 'Type is not allowed' });
+    }
+
+    const vacPayload = {
+      publishedAt: new Date(),
```

Comment:
Dit is geen string; zou dit niet iets van `new Date().toISOString()` moeten zijn?